Libraries/VirtualFunctions/VirtualFunctions.h

```diff
@@ -0,0 +1,312 @@
+// Code made by the RoBorregos team in 2020 for the RoboCup JR. Rescue Maze category.
+// Grecia Flores, Diego Prado, Marlon Romo, and Emérico Pedraza.
+//
+// This file contains functions that help to make virtual tests of the algorythm without the robot.
+// It cointains a char array simulating a physical map.
+// 'W' = wall.
+// 'B = black Tile.
+// 'R' = Robot's current position.
+// ' ' = Empty tile.
+// '.' = Just to fill spaces.
+// '1' = Low tile's difficulty.
+// '2' = Medium tile's difficulty.
+// '3' = High tile's difficulty.
+
+#include "Map.h"
+#include "Arduino.h"
+
+const uint8_t kNumberRows = 11, kNumberColumns = 11;
+uint8_t current_row = 1, current_column = 1;
+char virtual_map[11][11] = { {'.','W','.','W','.','W','.','W','.','W','.'},
+                            {'W','R','W','B','W',' ','W',' ','.','2','W'},
+                            {'.','.','.','.','.','.','.','.','.','.','.'},
+                            {'W',' ','.',' ','.',' ','.',' ','W',' ','W'},
+                            {'.','W','.','.','.','W','.','.','.','.','.'},
+                            {'W','B','W','3','W',' ','W',' ','.',' ','W'},
+                            {'.','.','.','.','.','W','.','W','.','.','.'},
+                            {'W',' ','.',' ','.','1','.','1','.',' ','W'},
+                            {'.','W','.','.','.','.','.','.','.','.','.'},
+                            {'W',' ','W',' ','W',' ','.',' ','W',' ','W'},
+                            {'.','W','.','W','.','W','.','W','.','W','.'} };
+
+// Function that prints the virtual map.
+void printVirtualMap() {
+  for(uint8_t row = 0 ; row < kNumberRows ; ++row) {
+    for(uint8_t column = 0 ; column < kNumberColumns ; ++column) {
+      Serial.print(virtual_map[row][column]);
+      if(column < kNumberColumns - 1) {
+        Serial.print(" ");
+      }
+    }
+    Serial.println("");
+  }
+}
+
+// Moves up the position of the robot on the virtual map.
```

Comment:
The comments of the functions goes only on the .h file, unless there's something extremely important that needs more detailing, then you can add it on the cpp file.